IoTHubBuddy/IoTViewer/IoTDataManager.cs

```diff
@@ -0,0 +1,228 @@
+                {
+                   return jsonObject;
+                } else
+                {
+                    return null;
+                }
+            }
+
+        }
+        /// <summary>
+        /// sends POST request to azure management REST API
+        /// </summary>
+        /// <param name="relative"></param>
+        /// <returns></returns>
+        public static async Task<JsonObject> PostIoTData(string relative)
+        {
+            string token = await AccountManager.GetTokenSilentlyAsync();
+            if(token == null)
+            {
+                return null;
+            }
+            using(var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri("https://management.azure.com/");
+                var message = new HttpRequestMessage(HttpMethod.Post, relative);
+
+                message.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+                message.Headers.AcceptLanguage.TryParseAdd("en-US");
+                message.Headers.Add("x-ms-version", "2013-11-01");
+                message.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+                var infoResult = await client.SendAsync(message);
+                string content = await infoResult.Content.ReadAsStringAsync();
+                var jsonObject = JsonObject.Parse(content);
+                if (!checkJsonObject(jsonObject))
+                {
+                    return null;
+                }
+                return jsonObject;
+            }
+            
+        }
+        /// <summary>
+        /// Get a list of azure subscriptions
+        /// </summary>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        public static async Task<ICollection<string>> GetSubscription(string token)
+        {
+            List<string> subscriptionIds = new List<string>();
+            string relative = "subscriptions?api-version=2015-01-01";
+            JsonObject result = await GetIoTData(token, relative);
+            if(result != null)
+            {
+                var subscriptions = result["value"].GetArray();
+                foreach (var subscription in subscriptions)
+                {
+                    string idContent = subscription.GetObject().GetNamedValue("subscriptionId").GetString();
+                    subscriptionIds.Add(idContent);
+                }
+            }
+            
+            return subscriptionIds;
+        }
+        /// <summary>
+        /// get a list of resource group for a specific azure subscription
+        /// </summary>
+        /// <param name="subId"></param>
+        /// <returns></returns>
+        public static async Task<ICollection<string>> GetResourceGroups(string subId)
+        {
+            List<string> groupNames = new List<string>();
+            var relative = "subscriptions/" + subId + "/resourcegroups?api-version=2015-01-01";
```

Comment:
Can we factor out the api version into a variable, for consistency and so it can be bumped in one place